### DISCUSSION TITLE
2008/09/09/%E5%BE%AA%E7%8E%AF%E8%BF%90%E8%A1%8C%E6%9F%90%E4%B8%AA%E6%97%B6%E9%97%B4%E5%90%8E%E9%80%80%E5%87%BA

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#242] [Announcements]
# 2008/09/09/%E5%BE%AA%E7%8E%AF%E8%BF%90%E8%A1%8C%E6%9F%90%E4%B8%AA%E6%97%B6%E9%97%B4%E5%90%8E%E9%80%80%E5%87%BA

有时候需要一个循环不是在迭代多少次之后停下，而是在运行多少时间之后停下。一般，直接的解决方案就是利用“时间计数器”函数，进入循环前，记下当前时间，然后每次循环迭代都查看一下当前时间，若超过所需时间，则退出循环。这个功能也可以使用“已用时间”Express VI来完成。

https://blog.qizhen.xyz/2008/09/09/%E5%BE%AA%E7%8E%AF%E8%BF%90%E8%A1%8C%E6%9F%90%E4%B8%AA%E6%97%B6%E9%97%B4%E5%90%8E%E9%80%80%E5%87%BA

**Comment by ruanqizhen** [comment]
之前博客下的留言：

Qizhen
2008年09月12日 10:40 编辑
“耗时40毫秒”这个VI只是用来演示的，所以也没有加error in/out第一和第三种方案在于第一种方案要多延时一个迭代周期，在这里就是40毫秒。比如一个极端的设置，我们设定0秒后循环停止。正确的行为就应当是“耗时40毫秒”这个VI一次都不执行。第三种方案可以做到，但是第一种就不行，它至少也要运行“耗时40毫秒”VI一次。当然，这几种方案一般都用在定时精度要求不高的应用场合，任务VI被多执行一次问题也不大，为了程序简介，用第一种方案也未尝不可。

回复
傻傻的
2008年09月12日 09:37 编辑
其实感觉第3中办法和第1种办法也没什么区别啊。一样是要等耗时40毫秒的子VI结束了才能结束。比如说设置的已用时间不是耗时的子VI的所用时间的整数倍的话。这两种方法就没什么区别了。反而感觉第一中方法来得干净利落，呵呵。个人见解，请大侠指点

回复
杰
2008年09月9日 19:46 编辑
耗时40ms.vi怎么没有error in and out呢？
没有错误处理的子VI不符合LV的编程规则，虽然程序没有问题，呵呵。
把这两个VI的错误簇输入并联到一起，大概就差不多了吧。不过搞不清到底LV会先执行哪个，所谓的 race condition 是吧？
但是本来，用软件来定时就不精确，虽然ms量级的效果还凑合。

回复
亮
2008年09月9日 18:14 编辑
其实可以直接把时间判定做在序列第一格,然后拉一条线从下面(不透过第二个序列)把结束与否传给循环终止就好.

回复
WL
2008年09月9日 17:47 编辑
学习

回复
五月的红枫
2008年09月9日 13:53 编辑
学习了．．．
 

回复
五月的红枫
2008年09月9日 13:53 编辑
沙发！
 

回复
